webapp/src/ts/modules/messages/messages-content.component.ts

```diff
@@ -0,0 +1,356 @@
+import { MarkReadService } from '@mm-services/mark-read.service';
+import { SendMessageService } from '@mm-services/send-message.service';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { SendMessageComponent } from '@mm-modals/send-message/send-message.component';
+
+/**
+*  In this context the URL parameter "id", can be:
+*    - the _id of the contact who is sending these messages
+*    - the phone number if it couldn't be resolved into a contact
+*    - the _id of the data_record if there is no discernible phone number
+*  This is determined by the URL parameter "type": 'contact', 'phone' or 'unknown' respectively.
+*/
+@Component({
+  selector: 'messages-content',
+  templateUrl: './messages-content.component.html'
+})
+export class MessagesContentComponent implements OnInit, OnDestroy, AfterViewInit, AfterViewChecked {
+  private userCtx;
+  private globalActions: GlobalActions;
+  messagesActions: MessagesActions;
+  loadingContent;
+  loadingMoreContent = false;
+  selectedConversation;
+  firstUnread;
+  send = { message: '' };
+  urlParameters = { type: '', id: '' };
+  subscriptions: Subscription = new Subscription();
+  textAreaFocused = false;
+  hasToScroll = false;
+  isAddRecipientBtnActive = false;
+  allLoaded = false;
+  checkScrollFnDef;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private route: ActivatedRoute,
+    private store: Store,
+    private changesService: ChangesService,
+    private messageContactService: MessageContactService,
+    private sessionService: SessionService,
+    private lineageModelGeneratorService: LineageModelGeneratorService,
+    private markReadService: MarkReadService,
+    private sendMessageService: SendMessageService,
+    private modalService: ModalService,
+  ) { }
+
+  ngOnInit(): void {
+    const selectorsSubscription = combineLatest(
+      this.store.pipe(select(Selectors.getSelectedConversation)),
+      this.store.pipe(select(Selectors.getLoadingContent)),
+    ).subscribe(([selectedConversation, loadingContent]) => {
+      this.selectedConversation = selectedConversation;
+      this.loadingContent = loadingContent;
+    });
+    this.subscriptions.add(selectorsSubscription);
+
+    this.globalActions = new GlobalActions(this.store);
+    this.messagesActions = new MessagesActions(this.store);
+    this.userCtx = this.sessionService.userCtx();
+    this.watchForChanges();
+  }
+
+  ngAfterViewInit(): void {
+    const routeSubscription = this.route.params.subscribe((params) => {
+      const [type, id] = params.type_id ? params.type_id.split(':') : [];
+      this.urlParameters.type = type;
+      this.urlParameters.id = id;
+      this.send.message = '';
+      this.hasToScroll = false;
+      this.allLoaded = false;
+      this.selectContact(this.urlParameters.id, this.urlParameters.type);
+      this.changeDetectorRef.detectChanges();
```

Comment:
Is this still needed? I tried removing it locally and messages loaded correctly.

Reply:
Yes, I tried to remove it but I'm getting the exception about change detection :( 
I've put it back